ChronoAPI/ChronoLog/include/errcode.h

```diff
@@ -0,0 +1,13 @@
+//
+// Created by kfeng on 12/19/22.
+//
+
+#ifndef CHRONOLOG_ERRCODE_H
+#define CHRONOLOG_ERRCODE_H
+
+#define CL_SUCCESS                  0
```

Comment:
Please check this. It should be consistent with true/false boolean values expected by the caller. Maybe CL_SUCCESS should be 1 (true)?

Reply:
In my current design, the caller should check if return value is equal to CL_SUCCESS for successful operation. All the APIs should return int. Positive numbers are reserved for returning updated ChronoTick after collision resolution in Record(). Negative numbers are left for all errors.

Reply:
What about functions that return boolean values and internally call functions that return error codes? Will the error code be typecast to boolean or will you add checks there?

Reply:
Do you mean all Client APIs return either true (SUCCESS) or false (FAIL) to the users and users can GetErrCode() to know what exact error is when they get a false return value? That means the Client library needs to store some states internally. I want to keep it state-less. Returning error code directly as the return value does not require any state memorization in Client library.

Reply:
I was referring to existing functions that return boolean values. You have to make sure it returns true on CL_SUCCESS a nd false otherwise if you don't throw an exception. You have to convert CL_SUCCESS which is of value 0 to boolean true in such functions.

Reply:
I have updated the Client APIs in https://github.com/scs-lab/ChronoLog/blob/a71d7a0702e4b796d9cd5944af4683753b14627f/Client/include/client.h. They all return int now.